pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -156,6 +161,12 @@ def _call_children_scripts(self):
         os.environ["NODE_RANK"] = str(self.cluster_environment.node_rank())
         os.environ["LOCAL_RANK"] = str(self.cluster_environment.local_rank())
 
+        # create a temporary directory used to synchronize processes on deadlock.
+        self._sync_dir = tempfile.mkdtemp()
+        os.environ["PL_DDP_SYNC_TMPDIR"] = self._sync_dir
+        if not os.path.exists(self._sync_dir):
+            os.makedirs(self._sync_dir)
```

Comment:
```suggestion
```

no need for this (mkdtemp() should already do this)

Reply:
Done !